File: app/src/main/java/ru/komcity/android/forum/ForumAdapter.java
package ru.komcity.android.forum;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.List;
import butterknife.BindView;
import butterknife.ButterKnife;
import ru.komcity.android.R;

public class ForumAdapter extends RecyclerView.Adapter<ForumAdapter.ForumViewHolder> {
    private List<Object> feedItemList;
    private Context mContext;
    private ForumClickListener onItemClickListener;

    public ForumAdapter(Context context, List<Object> feedItemList) {
        this.feedItemList = feedItemList;
        this.mContext = context;
    }

    @Override
    public ForumViewHolder onCreateViewHolder(ViewGroup viewGroup, int i) {
        View view = LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.forum_item, null);
        ForumViewHolder viewHolder = new ForumViewHolder(view);
        return viewHolder;
    }

    @Override
    public void onBindViewHolder(ForumViewHolder forumHolder, int i) {
        final ForumItem item = (ForumItem)feedItemList.get(i);

        //Setting text view title
        forumHolder.lbl_title_forum.setText(item.getTitle());
        forumHolder.lbl_replic_forum.setText(item.getCountReplic());
        forumHolder.lbl_theme_forum.setText(item.getCountTheme());
        forumHolder.lbl_description_forum_forum.setText(item.getDescription());

        View.OnClickListener listener = new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                onItemClickListener.onItemClick(item);
            }
        };
        forumHolder.lbl_title_forum.setOnClickListener(listener);
    }

    @Override
    public int getItemCount() {
        return (null != feedItemList ? feedItemList.size() : 0);
    }


    class ForumViewHolder extends RecyclerView.ViewHolder {
        @BindView(R.id.title_forum) TextView lbl_title_forum;
        @BindView(R.id.replic_forum)  TextView lbl_replic_forum;
        @BindView(R.id.theme_forum)  TextView lbl_theme_forum;
        @BindView(R.id.text_forum)  TextView lbl_description_forum_forum;

        public ForumViewHolder(View view) {
            super(view);
            ButterKnife.bind(this, view);
        }
    }

    public ForumClickListener getOnItemClickListener() {
        return onItemClickListener;
    }

    public void setOnItemClickListener(ForumClickListener onItemClickListener) {
        this.onItemClickListener = onItemClickListener;
    }
}

File: app/src/main/java/ru/komcity/mobile/ui/adapter/ForumAdapter.kt
package ru.komcity.mobile.ui.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import ru.komcity.mobile.databinding.ItemForumBinding
import ru.komcity.mobile.viewModel.ForumItem

/**
 * Created by Aleksey on 2020.02.28
 * <p>
 * Адаптер для списка форумов
 */
class ForumAdapter(private val items: List<ForumItem>,
                   private val listener: (title: String, forumName: String) -> Unit) : RecyclerView.Adapter<ForumAdapter.ViewHolder>() {
    private var _binding: ItemForumBinding? = null
    private val binding get() = _binding!!

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        _binding = ItemForumBinding.inflate(LayoutInflater.from(parent.context), parent, true)
        return ViewHolder(binding.root)
    }

    override fun getItemCount(): Int {
        return items.size
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.bind(items[position])
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        fun bind(item: ForumItem) {
            with(binding) {
                tvTitle.text = item.forumName
                tvThemeCount.text = item.countTheme
                tvReplicCount.text = item.countReplic
                tvDescription.text = item.description
                initClickListener(root, item)
            }
        }

        private fun initClickListener(itemView: View, item: ForumItem) {
            with(binding) {
                tvTitle.setOnClickListener {listener(item.forumName, item.linkForum)  }
                tvThemeCount.setOnClickListener {listener(item.forumName, item.linkForum)  }
                tvReplicCount.setOnClickListener {listener(item.forumName, item.linkForum)  }
                tvDescription.setOnClickListener {listener(item.forumName, item.linkForum)  }
                root.setOnClickListener { listener(item.forumName, item.linkForum) }
            }
        }
    }
}